### PR TITLE
[KYV-859] get/update-card-form-parameters endpoint and unit tests

### PR DESCRIPTION
**Pull request by laukkanenj**

### CODE REVIEW

File: paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentController.java [hunk 1]
```diff
@@ -193,4 +193,56 @@ public ResponseEntity<TreeMap<String, String>> getCardFormParameters(
...
     }
+
+    @GetMapping("/subscription/get/update-card-form-parameters")
+    public ResponseEntity<TreeMap<String, String>> getUpdateCardFormParameters(
+            @RequestParam(value = "merchantId") String merchantId,
+            @RequestParam(value = "namespace") String namespace,
+            @RequestParam(value = "orderId") String orderId
+    ) {
+        try {
+            String paytrailMerchantId = commonServiceConfigurationClient.getMerchantConfigurationValue(merchantId, namespace, ServiceConfigurationKeys.MERCHANT_PAYTRAIL_MERCHANT_ID);
```
Comment (severikupari1) [inline paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentController.java:204]: could these paytrail needed values be wrapped in some kind of  service that handles validation etc and return values.

Comment (severikupari1) [reply paymentapi/src/main/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentController.java:204]: Example can be found here, and even a good function to be used https://github.com/City-of-Helsinki/verkkokauppa-core/blob/10f046b7cd5ef004cc1e7688ca16108e34178445/paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilder.java#L33

File: paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContext.java [hunk 2]
```diff
@@ -22,4 +22,17 @@ public class PaytrailPaymentContext {
...
+    private String cardRedirectCancelUrl;
+    private String cardCallbackSuccessUrl;
+    private String cardCallbackCancelUrl;
+    private String updateCardRedirectSuccessUrl;
+    private String updateCardRedirectCancelUrl;
+    private String updateCardCallbackSuccessUrl;
+    private String updateCardCallbackCancelUrl;
+    private String redirectSuccessUrl;
+    private String redirectCancelUrl;
+    private String callbackSuccessUrl;
```
Comment (severikupari1) [inline paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContext.java:36]: Could you add some kinf of documentation to these values which does not have prefix ? Eq this is payment callback success url etc

Comment (severikupari1) [reply paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContext.java:36]: Or even better if you could rename these variables to have the prefix

File: paymentapi/src/main/resources/application-local.properties [hunk 3]
```diff
@@ -2,45 +2,42 @@ elasticsearch.service.url=localhost:9200
...
 serviceconfiguration.url=http://host.docker.internal:8187/serviceconfiguration/
 #server.port=8081
-
 logging.level.org.springframework.data.elasticsearch.client.WIRE=trace
-
 spring.activemq.broker-url=failover:(tcp://localhost:61616)
 spring.activemq.user=jmx
 spring.activemq.password=password
 spring.activemq.in-memory=false
-
```
Comment (severikupari1) [inline paymentapi/src/main/resources/application-local.properties:39]: Do not remove these spaces, they are there to keep some space between other values that are not in same "context"

File: paymentapi/src/test/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilderUnitTests.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+        ConfigurationDto mockConfigurationDto = new ConfigurationDto();
+        mockConfigurationDto.setKey(ServiceConfigurationKeys.MERCHANT_PAYTRAIL_MERCHANT_ID);
+        mockConfigurationDto.setValue(paytrailMerchantId);
+
+        ArrayList<ConfigurationDto> mockConfigurations = new ArrayList<>();
+        mockConfigurations.add(mockConfigurationDto);
+        mockMerchantDto.setConfigurations(mockConfigurations);
+
+        return mockMerchantDto;
```
Comment (severikupari1) [inline paymentapi/src/test/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilderUnitTests.java:144]: Could you add delete for this mockMerchant when test is run

**Review by severikupari1 — APPROVED** [review]
Some minor changes, very good automation tests keep it up!

File: paymentapi/src/test/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilderUnitTests.java (hunk 4 above)
Comment (laukkanenj) [reply paymentapi/src/test/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilderUnitTests.java:144]: It is not created into database. It is used for mocked call to  commonServiceConfigurationClient.getMerchantModel in https://github.com/City-of-Helsinki/verkkokauppa-core/blob/3234d3bebc156d5fb0622724ff283372ff0cffe9/paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/context/PaytrailPaymentContextBuilder.java#L64

**Pull request merged**